instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -0,0 +1,414 @@
+]
+
+
+class RequestInfo(typing.NamedTuple):
+    method: bytes
+    url: URL
+    headers: typing.Optional[Headers]
+    stream: typing.Optional[
+        typing.Union[httpx.SyncByteStream, httpx.AsyncByteStream]
+    ]
+    extensions: typing.Optional[dict]
+
+
+class ResponseInfo(typing.NamedTuple):
+    status_code: int
+    headers: typing.Optional[Headers]
+    stream: typing.Iterable[bytes]
+    extensions: typing.Optional[dict]
+
+
+def _get_default_span_name(method: str) -> str:
+    return "HTTP {}".format(method).strip()
+
+
+def _apply_status_code(span: Span, status_code: int) -> None:
+    if not span.is_recording():
+        return
+
+    span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, status_code)
+    span.set_status(Status(http_status_to_status_code(status_code)))
+
+
+def _prepare_attributes(method: bytes, url: URL) -> typing.Dict[str, str]:
+    _method = method.decode().upper()
+    _url = str(httpx.URL(url))
+    span_attributes = {
+        SpanAttributes.HTTP_METHOD: _method,
+        SpanAttributes.HTTP_URL: _url,
+    }
+    return span_attributes
+
+
+def _prepare_headers(headers: typing.Optional[Headers]) -> httpx.Headers:
+    return httpx.Headers(headers)
+
+
+class SyncOpenTelemetryTransport(httpx.BaseTransport):
+    """Sync transport class that will trace all requests made with a client.
+
+    Args:
+        transport: SyncHTTPTransport instance to wrap
+        tracer_provider: Tracer provider to use
+        request_hook: A hook that receives the span and request that is called
+            right after the span is created
+        response_hook: A hook that receives the span, request, and response
+            that is called right before the span ends
+    """
+
+    def __init__(
+        self,
+        transport: httpx.BaseTransport,
+        tracer_provider: typing.Optional[TracerProvider] = None,
+        request_hook: typing.Optional[RequestHook] = None,
+        response_hook: typing.Optional[ResponseHook] = None,
+    ):
+        self._transport = transport
+        self._tracer = get_tracer(
+            __name__,
+            instrumenting_library_version=__version__,
+            tracer_provider=tracer_provider,
+        )
+        self._request_hook = request_hook
+        self._response_hook = response_hook
+
+    def handle_request(
+        self,
+        method: bytes,
+        url: URL,
+        headers: typing.Optional[Headers] = None,
+        stream: typing.Optional[httpx.SyncByteStream] = None,
+        extensions: typing.Optional[dict] = None,
+    ) -> typing.Tuple[int, "Headers", httpx.SyncByteStream, dict]:
+        """Add request info to span."""
+        if context.get_value("suppress_instrumentation"):
+            return self._transport.handle_request(
+                method,
+                url,
+                headers=headers,
+                stream=stream,
+                extensions=extensions,
+            )
+
+        span_attributes = _prepare_attributes(method, url)
```

Comment:
Attributes should not be recorded if `is_recording()` is false.

Reply:
The attributes are not recorded in this function call, but later in `_apply_status_code`, which checks `is_recording()`. Do you suggest changing this to make it more clear?